content/knowledge-codemagic/build-inputs.md

```diff
@@ -0,0 +1,200 @@
+
+{{< highlight yaml "style=paraiso-dark">}}
+workflows:
+  ios:
+    inputs:
+      submitToTestFlight:
+        description: Enable testflight submission
+        type: boolean
+        default: false
+      runTests:
+        description: Run tests before build
+        type: boolean
+        default: true
+    integrations:
+      app_store_connect: MY_ASC_KEY
+    scripts:
+      - name: Run tests
+        script: xcode-project run-tests --project "project.xcodeproj" --scheme "App"
+        test_report: build/ios/test/*.xml
+        when:
+          condition: ${{ inputs.runTests }}
+      - ./setup_code_signing.sh
+      - xcode-project build-ipa --project "project.xcodeproj" --scheme "app"
+      - echo "Hello, ${{ inputs.name }}"
+    artifacts:
+      - build/**/*.ipa
+    publishing:
+      app_store_connect:
+        auth: integration
+        submit_to_testflight: ${{ inputs.submitToTestFlight }}
+{{< /highlight >}}
+
+In the above workflow user is prompted with two options when starting a build:
+1. whether to run tests before build, which controls the first script step using `when` condition,
+2. whether to submit the built ipa to TestFlight as part of App Store Connect publishing.
+
+Defaults are provided for both inputs and standard build can be started without choosing anything. 
+
+{{<notebox>}}
+**Note**: When using booleans in textual context, such as in scripts, truthy and falsy values are interpolated as strings `"true"` and `"false"` respectively. For example,
+```
+echo "My boolean: ${{ inputs.myTruthValue }}"
+```
+would be resolved to
+```
+echo: "My boolean: true"
+```
+if build is started with `myTruthValue: true`.
+{{</notebox>}}
+
+### Choice inputs
+
+Inputs with type choice provide a way to limit the user to choose only specific predefined values for inputs. Choice options are shown to user as a dropdowns. Choices can be useful to prevent typos or other human errors, and to easily present only the values that are relevant to use-case.
+
+Inputs with type `choice` must define additional field `options` where valid choices are listed.
+
+{{< highlight yaml "style=paraiso-dark">}}
+workflows:
+  ios:
+    inputs:
+      distributionType:
+        description: iOS distribution type 
+        type: choice
+        options: ["ad_hoc", "app_store", "development", "invalid"]
+        default: development
+    integrations:
+      app_store_connect: MY_ASC_KEY
+    environment:
+      ios_signing:
+        distribution_type: ${{ inputs.distributionType }}
+        bundle_identifier: com.example.app
+    scripts:
+      - xcode-project use-profiles
+      - flutter build ipa --debug --export-options-plist "${HOME:?}/export_options.plist"
+{{< /highlight >}}
+
```

Comment:
```suggestion
```

Reply:
Why would we not want an example of choice input?